src/coreclr/inc/gcinfodecoder.h

```diff
@@ -584,6 +588,7 @@ class TGcInfoDecoder
 #if defined(TARGET_ARM) || defined(TARGET_ARM64) || defined(TARGET_LOONGARCH64) || defined(TARGET_RISCV64)
     bool    HasTailCalls();
 #endif // TARGET_ARM || TARGET_ARM64 || TARGET_LOONGARCH64 || defined(TARGET_RISCV64)
+    ReturnKind GetReturnKind();
```

Comment:
```suggestion
#ifdef DECODE_OLD_FORMATS
        ReturnKind GetReturnKind();
#endif
```
Do we want these ifdefs to be more precise?

Reply:
I did not want the `DECODE_OLD_FORMATS` to creep out of the decoder into the dumper and beyond. I think there is no harm if `GetReturnKind` API to be present even if noone might call it. 
If we shim this further, then, at best, we can win an "int" field on the decoder, which would be a very small win, and that can still be done without shimming the interface. 

As long as we do not compute the return kind on hot paths and if asked for it returns `Invalid` (I saw that happen when I had bugs, it is useful) - I think it is a good level to `ifdef` the implementation, but not the API of the decoder.

Reply:
ok